weblate/vcs/git.py

```diff
@@ -564,6 +564,19 @@ def get_remote_configure(
             return (('remote "gerrit"', "url", push_url),)
         return (('remote "gerrit"', "url", None),)
 
+    def configure_remote(
+        self, pull_url: str, push_url: str, branch: str, fast: bool = True
+    ) -> None:
+        """
+        Configure remote repository.
+
+        Gets the gerrit username from push URL and
+        sets it as the value of gitreview.username.
+        """
+        gerrit_user = push_url.split("@")[0].split("//")[1]
```

Comment:
This needs to be more robust to handle all possible URL schemes. Your code will crash on SSH push URL like `git@github.com:WeblateOrg/weblate.git`.

See `parse_repo_url` method above in this file for an example that work well.

Reply:
Actually it even fails within the testsuite: https://github.com/WeblateOrg/weblate/actions/runs/8979415860/job/24661366078?pr=11585#step:7:4068

Reply:
I will improve this  to handle possible URL schemes. It was initially assumed that URLs coming from Gerrit will look like this:
ssh://<user>@<host>:<port>/<repo>.git

Actually, it has different URL schemes.

Reply:
AFAIK this can be any URL Git supports.